test/template.py

```diff
@@ -34,10 +34,12 @@ def exits(self):
 
     def random_path(self):
         exit_relay = random.choice(self.exits)
-        selected_relays = random.sample(self.routers, 3)
-        if exit_relay in selected_relays:
-            selected_relays.remove(exit_relay)
-        return selected_relays[0:2] + [exit_relay]
+        non_exits = [r for r in self.routers if 'exit' not in r.flags]
+        if len(non_exits) > 1:
+            return non_exits[0:2] + [exit_relay]
```

Comment:
I don't see a reason to only select the first two relays from the set of non-exits. Shouldn't those relays be selected from the full set of relays?